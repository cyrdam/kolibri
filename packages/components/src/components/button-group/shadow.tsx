import type { ButtonGroupProps } from '../../schema';
import type { JSX } from '@stencil/core';
import { Component, h, Host } from '@stencil/core';
import { KolButtonGroupWcTag } from '../../core/component-names';

@Component({
	tag: 'kol-button-group',
	styleUrls: {
		default: './style.scss',
	},
	shadow: true,
})
export class KolButtonGroup implements ButtonGroupProps {
	public render(): JSX.Element {
		return (
			<Host class="kol-button-group">
				<KolButtonGroupWcTag>
					<slot />
				</KolButtonGroupWcTag>
			</Host>
		);
	}
}
