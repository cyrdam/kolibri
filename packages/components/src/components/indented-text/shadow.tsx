import type { IndentedTextProps } from '../../schema';
import type { JSX } from '@stencil/core';
import { Component, h, Host } from '@stencil/core';
import { KolIndentedTextWcTag } from '../../core/component-names';

/**
 * @deprecated Will be removed in v3. Replace with custom HTML and styling.
 * @slot - Der Text.
 */
@Component({
	tag: 'kol-indented-text',
	styleUrls: {
		default: './style.scss',
	},
	shadow: true,
})
export class KolIndentedText implements IndentedTextProps {
	public render(): JSX.Element {
		return (
			<Host class="kol-indented-text">
				<KolIndentedTextWcTag>
					<slot />
				</KolIndentedTextWcTag>
			</Host>
		);
	}
}
