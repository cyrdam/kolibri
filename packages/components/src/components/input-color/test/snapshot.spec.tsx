import { executeTests } from 'stencil-awesome-test';

import { h } from '@stencil/core';
import { newSpecPage } from '@stencil/core/testing';

import { getInputColorHtml } from './html.mock';

import type { SpecPage } from '@stencil/core/testing';
import type { InputColorProps } from '../../../schema';
import { KolInputColor } from '../shadow';

executeTests<InputColorProps>(
	'InputColor',
	async (props): Promise<SpecPage> => {
		const page = await newSpecPage({
			components: [KolInputColor],
			template: () => <kol-input-color {...props} />,
		});
		return page;
	},
	{
		_label: ['Label'],
		_hideLabel: [true, false],
		_disabled: [true, false],
		_alert: [true, false],
		_icons: [[{ left: 'codicon codicon-home' }]],
		_touched: [true, false],
		_smartButton: [
			{
				_icons: ['codicon codicon-eye'],
				_hideLabel: true,
				_label: 'einblenden',
			},
		],
	},
	getInputColorHtml,
	{
		execMode: 'default', // ready
		needTimers: true,
	},
);
