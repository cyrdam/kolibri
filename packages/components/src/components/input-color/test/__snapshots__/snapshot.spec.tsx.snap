// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`kol-input-color should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _value="#FFF" _accessKey="V" 1`] = `
<kol-input-color _placeholder="Hier steht ein Platzhaltertext" _value="#FFF">
  <template shadowrootmode="open">
    <div class="kol-form-field kol-input-color" role="presentation">
      <label class="kol-form-field__label" htmlfor="id-nonce" id="id-nonce-label">
        <span class="kol-form-field__label-text">
          Label
          <span aria-hidden="true" class="access-key-hint">
            V
          </span>
        </span>
      </label>
      <div class="kol-form-field__input">
        <div class="kol-input-container">
          <div class="kol-input-container__container">
            <input accesskey="V" autocapitalize="off" autocomplete="off" autocorrect="off" class="kol-input" id="id-nonce" name="field" slot="input" type="color" value="#FFF">
          </div>
        </div>
      </div>
    </div>
  </template>
</kol-input-color>
`;

exports[`kol-input-color should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _value="#FFF" _alert=true 1`] = `
<kol-input-color _alert="" _placeholder="Hier steht ein Platzhaltertext" _value="#FFF">
  <template shadowrootmode="open">
    <div class="kol-form-field kol-input-color" role="presentation">
      <label class="kol-form-field__label" htmlfor="id-nonce" id="id-nonce-label">
        <span class="kol-form-field__label-text">
          <span>
            Label
          </span>
        </span>
      </label>
      <div class="kol-form-field__input">
        <div class="kol-input-container">
          <div class="kol-input-container__container">
            <input autocapitalize="off" autocomplete="off" autocorrect="off" class="kol-input" id="id-nonce" name="field" slot="input" type="color" value="#FFF">
          </div>
        </div>
      </div>
    </div>
  </template>
</kol-input-color>
`;

exports[`kol-input-color should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _value="#FFF" _disabled=true 1`] = `
<kol-input-color _placeholder="Hier steht ein Platzhaltertext" _value="#FFF">
  <template shadowrootmode="open">
    <div class="kol-form-field kol-form-field--disabled kol-input-color" role="presentation">
      <label class="kol-form-field__label" htmlfor="id-nonce" id="id-nonce-label">
        <span class="kol-form-field__label-text">
          <span>
            Label
          </span>
        </span>
      </label>
      <div class="kol-form-field__input">
        <div class="kol-input-container kol-input-container--disabled">
          <div class="kol-input-container__container">
            <input autocapitalize="off" autocomplete="off" autocorrect="off" class="kol-input kol-input--disabled" disabled="" id="id-nonce" name="field" slot="input" type="color" value="#FFF">
          </div>
        </div>
      </div>
    </div>
  </template>
</kol-input-color>
`;

exports[`kol-input-color should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _value="#FFF" _hint="Hint" 1`] = `
<kol-input-color _placeholder="Hier steht ein Platzhaltertext" _value="#FFF">
  <template shadowrootmode="open">
    <div class="kol-form-field kol-input-color" role="presentation">
      <label class="kol-form-field__label" htmlfor="id-nonce" id="id-nonce-label">
        <span class="kol-form-field__label-text">
          <span>
            Label
          </span>
        </span>
      </label>
      <div class="kol-form-field__input">
        <div class="kol-input-container">
          <div class="kol-input-container__container">
            <input aria-describedby="id-nonce-hint" autocapitalize="off" autocomplete="off" autocorrect="off" class="kol-input" id="id-nonce" name="field" slot="input" type="color" value="#FFF">
          </div>
        </div>
      </div>
      <span class="kol-form-field__hint" id="id-nonce-hint">
        Hint
      </span>
    </div>
  </template>
</kol-input-color>
`;

exports[`kol-input-color should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _value="#FFF" _icons={"left":{"icon":"codicon codicon-arrow-left"},"right":{"icon":"codicon codicon-arrow-right"}} 1`] = `
<kol-input-color _placeholder="Hier steht ein Platzhaltertext" _value="#FFF">
  <template shadowrootmode="open">
    <div class="kol-form-field kol-input-color" role="presentation">
      <label class="kol-form-field__label" htmlfor="id-nonce" id="id-nonce-label">
        <span class="kol-form-field__label-text">
          <span>
            Label
          </span>
        </span>
      </label>
      <div class="kol-form-field__input">
        <div class="kol-input-container">
          <div class="kol-input-container__adornment kol-input-container__adornment--start">
            <kol-icon _icons="codicon codicon-codicon codicon-arrow-left" _label=""></kol-icon>
          </div>
          <div class="kol-input-container__container">
            <input autocapitalize="off" autocomplete="off" autocorrect="off" class="kol-input" id="id-nonce" name="field" slot="input" type="color" value="#FFF">
          </div>
          <div class="kol-input-container__adornment kol-input-container__adornment--end">
            <kol-icon _icons="codicon codicon-codicon codicon-arrow-right" _label=""></kol-icon>
          </div>
        </div>
      </div>
    </div>
  </template>
</kol-input-color>
`;

exports[`kol-input-color should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _value="#FFF" _icons={"left":{"icon":"codicon codicon-arrow-left"}} 1`] = `
<kol-input-color _placeholder="Hier steht ein Platzhaltertext" _value="#FFF">
  <template shadowrootmode="open">
    <div class="kol-form-field kol-input-color" role="presentation">
      <label class="kol-form-field__label" htmlfor="id-nonce" id="id-nonce-label">
        <span class="kol-form-field__label-text">
          <span>
            Label
          </span>
        </span>
      </label>
      <div class="kol-form-field__input">
        <div class="kol-input-container">
          <div class="kol-input-container__adornment kol-input-container__adornment--start">
            <kol-icon _icons="codicon codicon-codicon codicon-arrow-left" _label=""></kol-icon>
          </div>
          <div class="kol-input-container__container">
            <input autocapitalize="off" autocomplete="off" autocorrect="off" class="kol-input" id="id-nonce" name="field" slot="input" type="color" value="#FFF">
          </div>
          <div class="kol-input-container__adornment kol-input-container__adornment--end"></div>
        </div>
      </div>
    </div>
  </template>
</kol-input-color>
`;

exports[`kol-input-color should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _value="#FFF" _icons={"right":{"icon":"codicon codicon-arrow-right"}} 1`] = `
<kol-input-color _placeholder="Hier steht ein Platzhaltertext" _value="#FFF">
  <template shadowrootmode="open">
    <div class="kol-form-field kol-input-color" role="presentation">
      <label class="kol-form-field__label" htmlfor="id-nonce" id="id-nonce-label">
        <span class="kol-form-field__label-text">
          <span>
            Label
          </span>
        </span>
      </label>
      <div class="kol-form-field__input">
        <div class="kol-input-container">
          <div class="kol-input-container__adornment kol-input-container__adornment--start"></div>
          <div class="kol-input-container__container">
            <input autocapitalize="off" autocomplete="off" autocorrect="off" class="kol-input" id="id-nonce" name="field" slot="input" type="color" value="#FFF">
          </div>
          <div class="kol-input-container__adornment kol-input-container__adornment--end">
            <kol-icon _icons="codicon codicon-codicon codicon-arrow-right" _label=""></kol-icon>
          </div>
        </div>
      </div>
    </div>
  </template>
</kol-input-color>
`;

exports[`kol-input-color should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _value="#FFF" _msg={"_type":"error","_description":"Es ist ein Fehler aufgetreten"} _hideMsg=true 1`] = `
<kol-input-color _hide-msg="" _placeholder="Hier steht ein Platzhaltertext" _value="#FFF">
  <template shadowrootmode="open">
    <div class="kol-form-field kol-form-field--hidden-msg kol-input-color" role="presentation">
      <label class="kol-form-field__label" htmlfor="id-nonce" id="id-nonce-label">
        <span class="kol-form-field__label-text">
          <span>
            Label
          </span>
        </span>
      </label>
      <div class="kol-form-field__input">
        <div class="kol-input-container">
          <div class="kol-input-container__container">
            <input aria-describedby="id-nonce-msg" autocapitalize="off" autocomplete="off" autocorrect="off" class="kol-input" id="id-nonce" name="field" slot="input" type="color" value="#FFF">
          </div>
        </div>
      </div>
    </div>
  </template>
</kol-input-color>
`;

exports[`kol-input-color should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _value="#FFF" _msg={"_type":"error","_description":"Es ist ein Fehler aufgetreten"} 1`] = `
<kol-input-color _placeholder="Hier steht ein Platzhaltertext" _value="#FFF">
  <template shadowrootmode="open">
    <div class="kol-form-field kol-input-color" role="presentation">
      <label class="kol-form-field__label" htmlfor="id-nonce" id="id-nonce-label">
        <span class="kol-form-field__label-text">
          <span>
            Label
          </span>
        </span>
      </label>
      <div class="kol-form-field__input">
        <div class="kol-input-container">
          <div class="kol-input-container__container">
            <input aria-describedby="id-nonce-msg" autocapitalize="off" autocomplete="off" autocorrect="off" class="kol-input" id="id-nonce" name="field" slot="input" type="color" value="#FFF">
          </div>
        </div>
      </div>
    </div>
  </template>
</kol-input-color>
`;

exports[`kol-input-color should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _value="#FFF" _readOnly=true 1`] = `
<kol-input-color _placeholder="Hier steht ein Platzhaltertext" _readonly="" _value="#FFF">
  <template shadowrootmode="open">
    <div class="kol-form-field kol-input-color" role="presentation">
      <label class="kol-form-field__label" htmlfor="id-nonce" id="id-nonce-label">
        <span class="kol-form-field__label-text">
          <span>
            Label
          </span>
        </span>
      </label>
      <div class="kol-form-field__input">
        <div class="kol-input-container">
          <div class="kol-input-container__container">
            <input autocapitalize="off" autocomplete="off" autocorrect="off" class="kol-input" id="id-nonce" name="field" slot="input" type="color" value="#FFF">
          </div>
        </div>
      </div>
    </div>
  </template>
</kol-input-color>
`;

exports[`kol-input-color should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _value="#FFF" _shortKey="S" 1`] = `
<kol-input-color _placeholder="Hier steht ein Platzhaltertext" _value="#FFF">
  <template shadowrootmode="open">
    <div class="kol-form-field kol-input-color" role="presentation">
      <label class="kol-form-field__label" htmlfor="id-nonce" id="id-nonce-label">
        <span class="kol-form-field__label-text">
          Label
          <span aria-hidden="true" class="access-key-hint">
            S
          </span>
        </span>
      </label>
      <div class="kol-form-field__input">
        <div class="kol-input-container">
          <div class="kol-input-container__container">
            <input autocapitalize="off" autocomplete="off" autocorrect="off" class="kol-input" id="id-nonce" name="field" slot="input" type="color" value="#FFF">
          </div>
        </div>
      </div>
    </div>
  </template>
</kol-input-color>
`;

exports[`kol-input-color should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _value="#FFF" _smartButton={"_icons":["codicon codicon-eye"],"_hideLabel":true,"_label":"einblenden"} 1`] = `
<kol-input-color _placeholder="Hier steht ein Platzhaltertext" _value="#FFF">
  <template shadowrootmode="open">
    <div class="kol-form-field kol-input-color" role="presentation">
      <label class="kol-form-field__label" htmlfor="id-nonce" id="id-nonce-label">
        <span class="kol-form-field__label-text">
          <span>
            Label
          </span>
        </span>
      </label>
      <div class="kol-form-field__input">
        <div class="kol-input-container">
          <div class="kol-input-container__adornment kol-input-container__adornment--start"></div>
          <div class="kol-input-container__container">
            <input autocapitalize="off" autocomplete="off" autocorrect="off" class="kol-input" id="id-nonce" name="field" slot="input" type="color" value="#FFF">
          </div>
          <div class="kol-input-container__adornment kol-input-container__adornment--end">
            <kol-button-wc _hidelabel="" _label="einblenden" tabindex="0"></kol-button-wc>
          </div>
        </div>
      </div>
    </div>
  </template>
</kol-input-color>
`;

exports[`kol-input-color should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _value="#FFF" _suggestions=["#F00","#0F0","#00F"] _accessKey="V" 1`] = `
<kol-input-color _placeholder="Hier steht ein Platzhaltertext" _value="#FFF">
  <template shadowrootmode="open">
    <div class="kol-form-field kol-input-color" role="presentation">
      <label class="kol-form-field__label" htmlfor="id-nonce" id="id-nonce-label">
        <span class="kol-form-field__label-text">
          Label
          <span aria-hidden="true" class="access-key-hint">
            V
          </span>
        </span>
      </label>
      <div class="kol-form-field__input">
        <div class="kol-input-container">
          <div class="kol-input-container__container">
            <input accesskey="V" autocapitalize="off" autocomplete="off" autocorrect="off" class="kol-input" id="id-nonce" list="id-nonce-list" name="field" slot="input" type="color" value="#FFF">
            <datalist id="id-nonce-list">
              <option value="#F00"></option>
              <option value="#0F0"></option>
              <option value="#00F"></option>
            </datalist>
          </div>
        </div>
      </div>
    </div>
  </template>
</kol-input-color>
`;

exports[`kol-input-color should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _value="#FFF" _suggestions=["#F00","#0F0","#00F"] _alert=true 1`] = `
<kol-input-color _alert="" _placeholder="Hier steht ein Platzhaltertext" _value="#FFF">
  <template shadowrootmode="open">
    <div class="kol-form-field kol-input-color" role="presentation">
      <label class="kol-form-field__label" htmlfor="id-nonce" id="id-nonce-label">
        <span class="kol-form-field__label-text">
          <span>
            Label
          </span>
        </span>
      </label>
      <div class="kol-form-field__input">
        <div class="kol-input-container">
          <div class="kol-input-container__container">
            <input autocapitalize="off" autocomplete="off" autocorrect="off" class="kol-input" id="id-nonce" list="id-nonce-list" name="field" slot="input" type="color" value="#FFF">
            <datalist id="id-nonce-list">
              <option value="#F00"></option>
              <option value="#0F0"></option>
              <option value="#00F"></option>
            </datalist>
          </div>
        </div>
      </div>
    </div>
  </template>
</kol-input-color>
`;

exports[`kol-input-color should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _value="#FFF" _suggestions=["#F00","#0F0","#00F"] _disabled=true 1`] = `
<kol-input-color _placeholder="Hier steht ein Platzhaltertext" _value="#FFF">
  <template shadowrootmode="open">
    <div class="kol-form-field kol-form-field--disabled kol-input-color" role="presentation">
      <label class="kol-form-field__label" htmlfor="id-nonce" id="id-nonce-label">
        <span class="kol-form-field__label-text">
          <span>
            Label
          </span>
        </span>
      </label>
      <div class="kol-form-field__input">
        <div class="kol-input-container kol-input-container--disabled">
          <div class="kol-input-container__container">
            <input autocapitalize="off" autocomplete="off" autocorrect="off" class="kol-input kol-input--disabled" disabled="" id="id-nonce" list="id-nonce-list" name="field" slot="input" type="color" value="#FFF">
            <datalist id="id-nonce-list">
              <option value="#F00"></option>
              <option value="#0F0"></option>
              <option value="#00F"></option>
            </datalist>
          </div>
        </div>
      </div>
    </div>
  </template>
</kol-input-color>
`;

exports[`kol-input-color should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _value="#FFF" _suggestions=["#F00","#0F0","#00F"] _hint="Hint" 1`] = `
<kol-input-color _placeholder="Hier steht ein Platzhaltertext" _value="#FFF">
  <template shadowrootmode="open">
    <div class="kol-form-field kol-input-color" role="presentation">
      <label class="kol-form-field__label" htmlfor="id-nonce" id="id-nonce-label">
        <span class="kol-form-field__label-text">
          <span>
            Label
          </span>
        </span>
      </label>
      <div class="kol-form-field__input">
        <div class="kol-input-container">
          <div class="kol-input-container__container">
            <input aria-describedby="id-nonce-hint" autocapitalize="off" autocomplete="off" autocorrect="off" class="kol-input" id="id-nonce" list="id-nonce-list" name="field" slot="input" type="color" value="#FFF">
            <datalist id="id-nonce-list">
              <option value="#F00"></option>
              <option value="#0F0"></option>
              <option value="#00F"></option>
            </datalist>
          </div>
        </div>
      </div>
      <span class="kol-form-field__hint" id="id-nonce-hint">
        Hint
      </span>
    </div>
  </template>
</kol-input-color>
`;

exports[`kol-input-color should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _value="#FFF" _suggestions=["#F00","#0F0","#00F"] _icons={"left":{"icon":"codicon codicon-arrow-left"},"right":{"icon":"codicon codicon-arrow-right"}} 1`] = `
<kol-input-color _placeholder="Hier steht ein Platzhaltertext" _value="#FFF">
  <template shadowrootmode="open">
    <div class="kol-form-field kol-input-color" role="presentation">
      <label class="kol-form-field__label" htmlfor="id-nonce" id="id-nonce-label">
        <span class="kol-form-field__label-text">
          <span>
            Label
          </span>
        </span>
      </label>
      <div class="kol-form-field__input">
        <div class="kol-input-container">
          <div class="kol-input-container__adornment kol-input-container__adornment--start">
            <kol-icon _icons="codicon codicon-codicon codicon-arrow-left" _label=""></kol-icon>
          </div>
          <div class="kol-input-container__container">
            <input autocapitalize="off" autocomplete="off" autocorrect="off" class="kol-input" id="id-nonce" list="id-nonce-list" name="field" slot="input" type="color" value="#FFF">
            <datalist id="id-nonce-list">
              <option value="#F00"></option>
              <option value="#0F0"></option>
              <option value="#00F"></option>
            </datalist>
          </div>
          <div class="kol-input-container__adornment kol-input-container__adornment--end">
            <kol-icon _icons="codicon codicon-codicon codicon-arrow-right" _label=""></kol-icon>
          </div>
        </div>
      </div>
    </div>
  </template>
</kol-input-color>
`;

exports[`kol-input-color should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _value="#FFF" _suggestions=["#F00","#0F0","#00F"] _icons={"left":{"icon":"codicon codicon-arrow-left"}} 1`] = `
<kol-input-color _placeholder="Hier steht ein Platzhaltertext" _value="#FFF">
  <template shadowrootmode="open">
    <div class="kol-form-field kol-input-color" role="presentation">
      <label class="kol-form-field__label" htmlfor="id-nonce" id="id-nonce-label">
        <span class="kol-form-field__label-text">
          <span>
            Label
          </span>
        </span>
      </label>
      <div class="kol-form-field__input">
        <div class="kol-input-container">
          <div class="kol-input-container__adornment kol-input-container__adornment--start">
            <kol-icon _icons="codicon codicon-codicon codicon-arrow-left" _label=""></kol-icon>
          </div>
          <div class="kol-input-container__container">
            <input autocapitalize="off" autocomplete="off" autocorrect="off" class="kol-input" id="id-nonce" list="id-nonce-list" name="field" slot="input" type="color" value="#FFF">
            <datalist id="id-nonce-list">
              <option value="#F00"></option>
              <option value="#0F0"></option>
              <option value="#00F"></option>
            </datalist>
          </div>
          <div class="kol-input-container__adornment kol-input-container__adornment--end"></div>
        </div>
      </div>
    </div>
  </template>
</kol-input-color>
`;

exports[`kol-input-color should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _value="#FFF" _suggestions=["#F00","#0F0","#00F"] _icons={"right":{"icon":"codicon codicon-arrow-right"}} 1`] = `
<kol-input-color _placeholder="Hier steht ein Platzhaltertext" _value="#FFF">
  <template shadowrootmode="open">
    <div class="kol-form-field kol-input-color" role="presentation">
      <label class="kol-form-field__label" htmlfor="id-nonce" id="id-nonce-label">
        <span class="kol-form-field__label-text">
          <span>
            Label
          </span>
        </span>
      </label>
      <div class="kol-form-field__input">
        <div class="kol-input-container">
          <div class="kol-input-container__adornment kol-input-container__adornment--start"></div>
          <div class="kol-input-container__container">
            <input autocapitalize="off" autocomplete="off" autocorrect="off" class="kol-input" id="id-nonce" list="id-nonce-list" name="field" slot="input" type="color" value="#FFF">
            <datalist id="id-nonce-list">
              <option value="#F00"></option>
              <option value="#0F0"></option>
              <option value="#00F"></option>
            </datalist>
          </div>
          <div class="kol-input-container__adornment kol-input-container__adornment--end">
            <kol-icon _icons="codicon codicon-codicon codicon-arrow-right" _label=""></kol-icon>
          </div>
        </div>
      </div>
    </div>
  </template>
</kol-input-color>
`;

exports[`kol-input-color should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _value="#FFF" _suggestions=["#F00","#0F0","#00F"] _msg={"_type":"error","_description":"Es ist ein Fehler aufgetreten"} _hideMsg=true 1`] = `
<kol-input-color _hide-msg="" _placeholder="Hier steht ein Platzhaltertext" _value="#FFF">
  <template shadowrootmode="open">
    <div class="kol-form-field kol-form-field--hidden-msg kol-input-color" role="presentation">
      <label class="kol-form-field__label" htmlfor="id-nonce" id="id-nonce-label">
        <span class="kol-form-field__label-text">
          <span>
            Label
          </span>
        </span>
      </label>
      <div class="kol-form-field__input">
        <div class="kol-input-container">
          <div class="kol-input-container__container">
            <input aria-describedby="id-nonce-msg" autocapitalize="off" autocomplete="off" autocorrect="off" class="kol-input" id="id-nonce" list="id-nonce-list" name="field" slot="input" type="color" value="#FFF">
            <datalist id="id-nonce-list">
              <option value="#F00"></option>
              <option value="#0F0"></option>
              <option value="#00F"></option>
            </datalist>
          </div>
        </div>
      </div>
    </div>
  </template>
</kol-input-color>
`;

exports[`kol-input-color should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _value="#FFF" _suggestions=["#F00","#0F0","#00F"] _msg={"_type":"error","_description":"Es ist ein Fehler aufgetreten"} 1`] = `
<kol-input-color _placeholder="Hier steht ein Platzhaltertext" _value="#FFF">
  <template shadowrootmode="open">
    <div class="kol-form-field kol-input-color" role="presentation">
      <label class="kol-form-field__label" htmlfor="id-nonce" id="id-nonce-label">
        <span class="kol-form-field__label-text">
          <span>
            Label
          </span>
        </span>
      </label>
      <div class="kol-form-field__input">
        <div class="kol-input-container">
          <div class="kol-input-container__container">
            <input aria-describedby="id-nonce-msg" autocapitalize="off" autocomplete="off" autocorrect="off" class="kol-input" id="id-nonce" list="id-nonce-list" name="field" slot="input" type="color" value="#FFF">
            <datalist id="id-nonce-list">
              <option value="#F00"></option>
              <option value="#0F0"></option>
              <option value="#00F"></option>
            </datalist>
          </div>
        </div>
      </div>
    </div>
  </template>
</kol-input-color>
`;

exports[`kol-input-color should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _value="#FFF" _suggestions=["#F00","#0F0","#00F"] _readOnly=true 1`] = `
<kol-input-color _placeholder="Hier steht ein Platzhaltertext" _readonly="" _value="#FFF">
  <template shadowrootmode="open">
    <div class="kol-form-field kol-input-color" role="presentation">
      <label class="kol-form-field__label" htmlfor="id-nonce" id="id-nonce-label">
        <span class="kol-form-field__label-text">
          <span>
            Label
          </span>
        </span>
      </label>
      <div class="kol-form-field__input">
        <div class="kol-input-container">
          <div class="kol-input-container__container">
            <input autocapitalize="off" autocomplete="off" autocorrect="off" class="kol-input" id="id-nonce" list="id-nonce-list" name="field" slot="input" type="color" value="#FFF">
            <datalist id="id-nonce-list">
              <option value="#F00"></option>
              <option value="#0F0"></option>
              <option value="#00F"></option>
            </datalist>
          </div>
        </div>
      </div>
    </div>
  </template>
</kol-input-color>
`;

exports[`kol-input-color should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _value="#FFF" _suggestions=["#F00","#0F0","#00F"] _shortKey="S" 1`] = `
<kol-input-color _placeholder="Hier steht ein Platzhaltertext" _value="#FFF">
  <template shadowrootmode="open">
    <div class="kol-form-field kol-input-color" role="presentation">
      <label class="kol-form-field__label" htmlfor="id-nonce" id="id-nonce-label">
        <span class="kol-form-field__label-text">
          Label
          <span aria-hidden="true" class="access-key-hint">
            S
          </span>
        </span>
      </label>
      <div class="kol-form-field__input">
        <div class="kol-input-container">
          <div class="kol-input-container__container">
            <input autocapitalize="off" autocomplete="off" autocorrect="off" class="kol-input" id="id-nonce" list="id-nonce-list" name="field" slot="input" type="color" value="#FFF">
            <datalist id="id-nonce-list">
              <option value="#F00"></option>
              <option value="#0F0"></option>
              <option value="#00F"></option>
            </datalist>
          </div>
        </div>
      </div>
    </div>
  </template>
</kol-input-color>
`;

exports[`kol-input-color should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _value="#FFF" _suggestions=["#F00","#0F0","#00F"] _smartButton={"_icons":["codicon codicon-eye"],"_hideLabel":true,"_label":"einblenden"} 1`] = `
<kol-input-color _placeholder="Hier steht ein Platzhaltertext" _value="#FFF">
  <template shadowrootmode="open">
    <div class="kol-form-field kol-input-color" role="presentation">
      <label class="kol-form-field__label" htmlfor="id-nonce" id="id-nonce-label">
        <span class="kol-form-field__label-text">
          <span>
            Label
          </span>
        </span>
      </label>
      <div class="kol-form-field__input">
        <div class="kol-input-container">
          <div class="kol-input-container__adornment kol-input-container__adornment--start"></div>
          <div class="kol-input-container__container">
            <input autocapitalize="off" autocomplete="off" autocorrect="off" class="kol-input" id="id-nonce" list="id-nonce-list" name="field" slot="input" type="color" value="#FFF">
            <datalist id="id-nonce-list">
              <option value="#F00"></option>
              <option value="#0F0"></option>
              <option value="#00F"></option>
            </datalist>
          </div>
          <div class="kol-input-container__adornment kol-input-container__adornment--end">
            <kol-button-wc _hidelabel="" _label="einblenden" tabindex="0"></kol-button-wc>
          </div>
        </div>
      </div>
    </div>
  </template>
</kol-input-color>
`;

exports[`kol-input-color should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _value="#FFF" _suggestions=["#F00","#0F0","#00F"] _touched=true 1`] = `
<kol-input-color _placeholder="Hier steht ein Platzhaltertext" _touched="" _value="#FFF">
  <template shadowrootmode="open">
    <div class="kol-form-field kol-form-field--touched kol-input-color" role="presentation">
      <label class="kol-form-field__label" htmlfor="id-nonce" id="id-nonce-label">
        <span class="kol-form-field__label-text">
          <span>
            Label
          </span>
        </span>
      </label>
      <div class="kol-form-field__input">
        <div class="kol-input-container">
          <div class="kol-input-container__container">
            <input autocapitalize="off" autocomplete="off" autocorrect="off" class="kol-input kol-input--touched" id="id-nonce" list="id-nonce-list" name="field" slot="input" type="color" value="#FFF">
            <datalist id="id-nonce-list">
              <option value="#F00"></option>
              <option value="#0F0"></option>
              <option value="#00F"></option>
            </datalist>
          </div>
        </div>
      </div>
    </div>
  </template>
</kol-input-color>
`;

exports[`kol-input-color should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _value="#FFF" _suggestions=["#F00","#0F0","#00F"] 1`] = `
<kol-input-color _placeholder="Hier steht ein Platzhaltertext" _value="#FFF">
  <template shadowrootmode="open">
    <div class="kol-form-field kol-input-color" role="presentation">
      <label class="kol-form-field__label" htmlfor="id-nonce" id="id-nonce-label">
        <span class="kol-form-field__label-text">
          <span>
            Label
          </span>
        </span>
      </label>
      <div class="kol-form-field__input">
        <div class="kol-input-container">
          <div class="kol-input-container__container">
            <input autocapitalize="off" autocomplete="off" autocorrect="off" class="kol-input" id="id-nonce" list="id-nonce-list" name="field" slot="input" type="color" value="#FFF">
            <datalist id="id-nonce-list">
              <option value="#F00"></option>
              <option value="#0F0"></option>
              <option value="#00F"></option>
            </datalist>
          </div>
        </div>
      </div>
    </div>
  </template>
</kol-input-color>
`;

exports[`kol-input-color should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _value="#FFF" _touched=true 1`] = `
<kol-input-color _placeholder="Hier steht ein Platzhaltertext" _touched="" _value="#FFF">
  <template shadowrootmode="open">
    <div class="kol-form-field kol-form-field--touched kol-input-color" role="presentation">
      <label class="kol-form-field__label" htmlfor="id-nonce" id="id-nonce-label">
        <span class="kol-form-field__label-text">
          <span>
            Label
          </span>
        </span>
      </label>
      <div class="kol-form-field__input">
        <div class="kol-input-container">
          <div class="kol-input-container__container">
            <input autocapitalize="off" autocomplete="off" autocorrect="off" class="kol-input kol-input--touched" id="id-nonce" name="field" slot="input" type="color" value="#FFF">
          </div>
        </div>
      </div>
    </div>
  </template>
</kol-input-color>
`;

exports[`kol-input-color should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _value="#FFF" 1`] = `
<kol-input-color _placeholder="Hier steht ein Platzhaltertext" _value="#FFF">
  <template shadowrootmode="open">
    <div class="kol-form-field kol-input-color" role="presentation">
      <label class="kol-form-field__label" htmlfor="id-nonce" id="id-nonce-label">
        <span class="kol-form-field__label-text">
          <span>
            Label
          </span>
        </span>
      </label>
      <div class="kol-form-field__input">
        <div class="kol-input-container">
          <div class="kol-input-container__container">
            <input autocapitalize="off" autocomplete="off" autocorrect="off" class="kol-input" id="id-nonce" name="field" slot="input" type="color" value="#FFF">
          </div>
        </div>
      </div>
    </div>
  </template>
</kol-input-color>
`;
